### PR TITLE
CFE-4313: Adjusted package_methods generic and yum to allow for packages with dashes

### PR DESCRIPTION
**Pull request by craigcomstock**
Trying to install python3 would result in attempting to install all python3-*.* packages which includes hundreds of python modules.

Ticket: CFE-4313
Changelog: title

### CODE REVIEW

File: lib/packages.cf [hunk 1]
```diff
@@ -1003,7 +1003,7 @@ body package_method yum
       package_list_arch_regex    => "$(rpm_knowledge.rpm3_arch_regex)";
 
       package_installed_regex => ".*";
-      package_name_convention => "$(name)-$(version).$(arch)";
+      package_name_convention => "$(name)";
```
Comment (nickanderson) [inline lib/packages.cf:1006]: Wouldn't this result in being unable to specify the version and/or architecture for a package?

**Review by nickanderson — COMMENTED** [review]
hummm

File: lib/packages.cf (hunk 1 above)
Comment (craigcomstock) [reply lib/packages.cf:1006]: Likely a better fix would be to change the C code to anchor the `$(version).$(arch)` portion to the end of the name. I can investigate that instead.

**Comment by craigcomstock** [comment]
I took a peek at the C code in core/cf-agent/verify_packages.c and decided it's too complex for a quick fix. So will close and leave CFE-4313 open/todo.

**Pull request closed without merging**